### PR TITLE
kernel/semaphore: Initialize kernel semaphores with sem_init instead of SEM_INITIALIZER

### PR DESCRIPTION
**Pull request by jeongarmy**
All kernel semaphores should be intialized by sem_init for recovery.
In App separation, they are registered to a list in sem_init and recovered when fault occurs.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [1a98c59fe7cb18f19fe448896049bb0d8ab0573e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1a98c59fe7cb18f19fe448896049bb0d8ab0573e] - Code Rule Check OK.

File: os/kernel/init/os_bringup.c [hunk 1]
```diff
@@ -272,6 +272,10 @@ static inline void os_do_appstart(void)
 	task_manager_drv_register();
 #endif
 
+#ifdef CONFIG_MESSAGING_IPC
+	messaging_initialize();
```
Comment (jeongchanKim) [inline os/kernel/init/os_bringup.c]: I think this is not related with os_bringup. How about moving this to preapp init?

**Comment by seinfra** [comment]
Target : [6f5426a2dd396eacaeb97d4bfc33a621c334ed38] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6f5426a2dd396eacaeb97d4bfc33a621c334ed38] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [21fd164c3f6e4bdab881c3ebee1e76f39fffa366] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [21fd164c3f6e4bdab881c3ebee1e76f39fffa366] - Code Rule Check (C++) OK.

File: os/drivers/net/phy_notify.c [hunk 2]
```diff
@@ -334,6 +334,26 @@ static int phy_handler_3(int irq, FAR void *context, FAR void *arg)
...
+ *
+ * Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+int phy_notify_initialize(void)
```
Comment (sunghan-chang) [inline os/drivers/net/phy_notify.c]: Who calls this?

File: os/drivers/syslog/ramlog.c [hunk 3]
```diff
@@ -678,6 +678,24 @@ int ramlog_consoleinit(void)
...
+#ifndef CONFIG_RAMLOG_NONBLOCKING
+	if ((priv->rl_waitsem.flags & FLAGS_INITIALIZED) == 0) {
+
+		sem_init(&priv->rl_waitsem, 0, 0);
+
+		/*
+		 * The rl_waitsem semaphore is used for signaling and, hence,
+		 * should not have priority inheritance enabled.
+		 */
+		sem_setprotocol(&priv->rl_waitsem, SEM_PRIO_NONE);
```
Comment (sunghan-chang) [inline os/drivers/syslog/ramlog.c]: In original codes, there is no set of protocol type. Is this really indeed? Then, please split the commits for this.

File: os/include/semaphore.h [hunk 4]
```diff
@@ -141,6 +141,11 @@ struct sem_s {
 typedef struct sem_s sem_t;
 
 /* Initializers */
+/* NOTE : It should NOT be used in kernel regions because it doesn't call sem_init.
```
Comment (sunghan-chang) [inline os/include/semaphore.h]: in kernel space

File: os/kernel/init/os_bringup.c (hunk 1 above)
Comment (sunghan-chang) [reply os/kernel/init/os_bringup.c]: The preapp_init is in user space so that you should not move there.

Comment (sunghan-chang) [inline os/kernel/init/os_bringup.c]: needs including the header for prototype.

**Comment by seinfra** [comment]
Target : [0eb23062a39725de3771794ce9912b1d2ad275bc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0eb23062a39725de3771794ce9912b1d2ad275bc] - Code Rule Check OK.

File: os/drivers/syslog/ramlog.c (hunk 3 above)
Comment (sunghan-chang) [reply os/drivers/syslog/ramlog.c]: Don't you need this?

Comment (jeongarmy) [reply os/drivers/syslog/ramlog.c]: It's needed because this semaphore is used for signaling. I updated with another commit.

**Comment by seinfra** [comment]
Target : [92b97a4730175bacd3295c477233531b6f43f2cc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [92b97a4730175bacd3295c477233531b6f43f2cc] - Code Rule Check OK.
